### PR TITLE
chore: fixing OMP issues for macOS ARM and Python3.11.

### PR DESCRIPTION
**Pull request by bcm-at-zama**
- Looks like the OMP issues also happen on ARM
- And by the way, we didn't have the fix for Python3.11

### CODE REVIEW

File: script/make_utils/fix_omp_issues_for_intel_mac.sh [hunk 1]
```diff
@@ -10,7 +10,7 @@ MACHINE=$(uname -m)
 PYTHON_VERSION=$(python --version | cut -d' ' -f2 | cut -d'.' -f1,2)
 DO_REGENERATE=0
 
-if [ "$UNAME" == "Darwin" ] && [ "$MACHINE" != "arm64" ]
```
Comment (bcm-at-zama) [inline script/make_utils/fix_omp_issues_for_intel_mac.sh:13]: I don't know how it works for you, @fd0r @RomanBredehoft 
Could you tell me?

File: script/make_utils/fix_omp_issues_for_intel_mac.sh [hunk 2]
```diff
@@ -63,6 +63,15 @@ then
         ln -s "${WHICH_VENV}"/lib/"${WHICH_PYTHON}"/site-packages/concrete/.dylibs/libomp.dylib "${WHICH_VENV}"/lib/"${WHICH_PYTHON}"/./site-packages/sklearn/.dylibs/libomp.dylib
         rm "${WHICH_VENV}"/lib/"${WHICH_PYTHON}"/./site-packages/functorch/.dylibs/libiomp5.dylib
         ln -s "${WHICH_VENV}"/lib/"${WHICH_PYTHON}"/site-packages/concrete/.dylibs/libomp.dylib "${WHICH_VENV}"/lib/"${WHICH_PYTHON}"/./site-packages/functorch/.dylibs/libiomp5.dylib
+
+    elif [ "$PYTHON_VERSION" == "3.11" ]
+    then
+        rm "${WHICH_VENV}"/lib/"${WHICH_PYTHON}"/./site-packages/xgboost/.dylibs/libomp.dylib
+        ln -s "${WHICH_VENV}"/lib/"${WHICH_PYTHON}"/site-packages/concrete/.dylibs/libomp.dylib "${WHICH_VENV}"/lib/"${WHICH_PYTHON}"/./site-packages/xgboost/.dylibs/libomp.dylib
```
Comment (bcm-at-zama) [inline script/make_utils/fix_omp_issues_for_intel_mac.sh:70]: completely trivial to add now, thanks to the `DO_REGENERATE=1` option!
we'll be able to add 3.12 easily

File: script/make_utils/fix_omp_issues_for_intel_mac.sh (hunk 1 above)
Comment (RomanBredehoft) [reply script/make_utils/fix_omp_issues_for_intel_mac.sh:13]: I'm getting an error when running `make sybc_env`

File: script/make_utils/fix_omp_issues_for_intel_mac.sh [hunk 3]
```diff
@@ -41,31 +41,42 @@ then
...
-        ln -s "${WHICH_VENV}"/lib/"${WHICH_PYTHON}"/site-packages/concrete/.dylibs/libomp.dylib "${WHICH_VENV}"/lib/"${WHICH_PYTHON}"/./site-packages/torch/lib/libiomp5.dylib
-        rm "${WHICH_VENV}"/lib/"${WHICH_PYTHON}"/./site-packages/sklearn/.dylibs/libomp.dylib
-        ln -s "${WHICH_VENV}"/lib/"${WHICH_PYTHON}"/site-packages/concrete/.dylibs/libomp.dylib "${WHICH_VENV}"/lib/"${WHICH_PYTHON}"/./site-packages/sklearn/.dylibs/libomp.dylib
-        rm "${WHICH_VENV}"/lib/"${WHICH_PYTHON}"/./site-packages/functorch/.dylibs/libiomp5.dylib
-        ln -s "${WHICH_VENV}"/lib/"${WHICH_PYTHON}"/site-packages/concrete/.dylibs/libomp.dylib "${WHICH_VENV}"/lib/"${WHICH_PYTHON}"/./site-packages/functorch/.dylibs/libiomp5.dylib
     else
-        echo "Please have a look to libraries libiomp5.dylib related to torch and then"
-        echo "apply appropriate fix"
-        exit 255
+        # The error is specific to python version
```
Comment (bcm-at-zama) [inline script/make_utils/fix_omp_issues_for_intel_mac.sh:55]: So finally I haven't changed the things for Intel, and I have created a new category for ARM, where all python are equivalent

File: script/make_utils/fix_omp_issues_for_intel_mac.sh (hunk 1 above)
Comment (bcm-at-zama) [reply script/make_utils/fix_omp_issues_for_intel_mac.sh:13]: @RomanBredehoft : do you still get an error and which error please

**Review by andrei-stoian-zama — APPROVED** [review]
@RomanBredehoft said it works!

**Pull request merged**